### PR TITLE
Implement moving pointing arrow.

### PR DESCRIPTION
**Pull request by pointer-to-null**
## About the PR 

The pointing arrow now fly from the pointer (player) to the pointed location. This makes intentions more clear.

The speed at which the arrow moves depends on the distance. The further it goes, the faster.

The flying part can be completely disabled if `isFlying` variable is set to false before `PointingArrowComponent` startup.

**Screenshots**

https://user-images.githubusercontent.com/91910481/143766401-b6040c17-a50d-47bf-92d3-b7da1eccbc28.mp4

**Changelog**

:cl:
- add: Now the pointing arrow actually moves from the player to the pointed location.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
I've seen unitystation's version of this and I wasn't a fan, IMO.

**Comment by pointer-to-null** [comment]
This helps determine who is pointing where, alot. Especially when there are several people hanging around.

**Comment by moonheart08** [comment]
@metalgearsloth this seems like a good thing to make an option in the options menu

**Comment by PaulRitter** [comment]
a clientside-option for this should suffice, yeah

**Comment by mirrorcult** [comment]
personally i'm fine with this

File: Content.Server/Pointing/EntitySystems/PointingSystem.cs [hunk 1]
```diff
@@ -46,6 +46,8 @@ internal sealed class PointingSystem : EntitySystem
 
         private const float PointingRange = 15f;
 
+        private MapCoordinates playerCoordinates;
```
Comment (mirrorcult) [inline Content.Server/Pointing/EntitySystems/PointingSystem.cs:49]: this being on the serverside system means you can't have more than one person pointing at the same time

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by Zumorica** [comment]
My only issue with this is that pointing arrows are entirely server-side right now, wasting a lot of bandwidth for no good reason...
But refactoring them to be client-side only is honestly out of scope for this PR. I like that animation, personally

**Comment by DogZeroX** [comment]
Could probably have a limit of how many arrows you can "shoot"

**Comment by metalgearsloth** [comment]
> a clientside-option for this should suffice, yeah

Should be server-side and replicated because people with it off are at an inherent disadvantage.

File: Content.Server/Pointing/EntitySystems/PointingSystem.cs [hunk 2]
```diff
@@ -210,10 +215,21 @@ private void AddPointingVerb(GetOtherVerbsEvent args)
...
         }
 
+        private void OnPointingArrowCompStartup(EntityUid uid, PointingArrowComponent component, ComponentStartup args)
+        {
+            if(component._isFlying)
+            {
+                component._pointerCoord = playerCoordinates;
+                component._pointedCoord = EntityManager.GetEntity(uid).Transform.MapPosition;
+                //Sets the initial arrow position to the player coordinates.
+                EntityManager.GetEntity(uid).Transform.WorldPosition = playerCoordinates.Position;
```
Comment (metalgearsloth) [inline Content.Server/Pointing/EntitySystems/PointingSystem.cs:229]: Don't use GetEntity, just use GetComponent

File: Content.Server/Pointing/EntitySystems/PointingSystem.cs [hunk 3]
```diff
@@ -226,7 +242,43 @@ public override void Update(float frameTime)
         {
             foreach (var component in EntityManager.EntityQuery<PointingArrowComponent>())
             {
-                component.Update(frameTime);
+                if(component._isFlying)
+                {
+                    var directionStep = component._pointedCoord.Position - component._pointerCoord.Position;
+                    component.Owner.Transform.WorldPosition += directionStep * frameTime * component._flyingSpeed;
```
Comment (metalgearsloth) [inline Content.Server/Pointing/EntitySystems/PointingSystem.cs:248]: Don't use the transform property and as mirror said don't use mapcoordinates.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by pointer-to-null** [comment]
All code reviewers comments and serverside option are noted.

> My only issue with this is that pointing arrows are entirely server-side right now, wasting a lot of bandwidth for no good reason... But refactoring them to be client-side only is honestly out of scope for this PR. I like that animation, personally

This is good point actually. So, I think the best decision would be to hold this PR for now. And implement this feature later.

In this regard, I have a question. Should I convert this PR to draft or close it?

**Pull request closed without merging**